cmd/opampsupervisor/specification/README.md

```diff
@@ -220,19 +225,69 @@ telemetry:
 
 ```
 
-**Note:**
+#### Notes on `agent::config_files`, `agent::args`, and `agent::env`
+
+Please be aware that when using the `agent::config_files` parameter,
+the configuration files specified are applied in the order they are specified.
+In other words, configuration files are merged from the top of the list to the bottom.
+Configuration added by files at the top of the list may be overwritten by the later ones.
+
+The indicated configuration files are merged in memory and the resulting configuration
+is written to `<storage::directory>/effective.yaml`.
 
-Please be aware that when using the `.agent.config_files` parameter,
-the configuration files specified are applied after the configuration from the OpAMP server.
-After the configuration files, arguments present in `.agent.args` are passed to the executable binary.
-The environmanet variables specified in `.agent.env` are set in the collector process environment.
+There are a few "magic" configuration files that can be used to completely
```

Comment:
I vote we do not refer to these as `magic` configuration files. Something like `reserved` or `key` or `global`, `variable` or `special` as it feels more professional. I dont want users to think using these variables are something to be avoided.